### PR TITLE
Issue #6907 Added ability to prevent LDAP org role resync

### PR DESCRIPTION
**Pull request by brandonraphael**
Added option in pkg/login/ldap.go, and pkg/settting/setting.go to prevent changes to user role on login, fixing issue #6907 based on 'allow_role_sync' value in defaults.ini

### CODE REVIEW

**Comment by torkelo** [comment]
not sure this is a good way to do it. The initial role is from ldap group mappings but then you can change it in Grafana. I think either disable the ldap group -> org member & role sync all together or not at all, this in between option seems problematic.

**Comment by brandonraphael** [comment]
We wanted to leave the initial functionality untouched, while still offering the ability to easily disable role sync. The case of removing ldap group -> org member role sync altogether is even simpler than this setup, and just involves changing group.OrgRole to org.Role as suggested in #6907 

If you don't believe this is necessary, we can close this pull request, but we thought this was an easy solution to a common problem, and was worth implementing.

**Comment by torkelo** [comment]
The thing I am worried about with this solution is the in between nature of what is authoritative, what determines user role, is it ldap group or is it grafana db? This mixes it, so the initial login is ldap group, then it what ever is set in grafana db. So it becomes a bit inconsistent

**Comment by brandonraphael** [comment]
So should we close the pull request? Or are there any changes you would like made?

File: pkg/login/ldap.go [hunk 1]
```diff
@@ -260,9 +260,18 @@ func (a *ldapAuther) SyncOrgRoles(user *m.User, ldapUser *LdapUserInfo) error {
...
-						return err
+					if setting.LdapAllowRoleSync {
+						// allow sync of org roles
+						cmd := m.UpdateOrgUserCommand{OrgId: org.OrgId, UserId: user.Id, Role: group.OrgRole}
+						if err := bus.Dispatch(&cmd); err != nil {
+							return err
+						}
+					} else {
+						// disallow sync of org roles
+						cmd := m.UpdateOrgUserCommand{OrgId: org.OrgId, UserId: user.Id, Role: org.Role}
```
Comment (torkelo) [inline pkg/login/ldap.go]: why is this called when LdapAllowRoleSync is disabled? Should it not just return (do nothing) ?

Comment (brandonraphael) [inline pkg/login/ldap.go]: Yes, you are right.

**Comment by haidaraM** [comment]
Will this feature be avalaible in the next release of Grafana (4.3.0) ? I just checked the release notes of 4.3.0-beta1 but I see nothing about that.

**Comment by torkelo** [comment]
closing this due to lack interest in the feature request (https://github.com/grafana/grafana/issues/6907), can open again if there is more interest

**Pull request closed without merging**